pgmpy/utils/utils.py

```diff
@@ -381,14 +383,10 @@ def preprocess_data(df):
     df = df.copy()
     dtypes = {}
     for col in df.columns:
-        if pd.api.types.is_integer_dtype(df[col]):
```

Comment:
@mhmohona Is the change in this block necessary? I don't think lines 409-413 has the same behavior as here, for example, does line 413 preserve the ordering of the categories?

Reply:
Sorry, this line is mistakenly removed.

Reply:
Sorry, this line has been mistakenly edited. I was playing around with some other functions and forgot to bring it back.

Reply:
Yes, this change is not relevant. I changed things here and there and forgot to bring them back.